src/backups.py

```diff
@@ -411,6 +411,9 @@ def _initialise_stanza(self) -> None:
 
     @property
     def _is_primary_pgbackrest_service_running(self) -> bool:
+        if not self.charm.primary_endpoint:
+            logger.warning("Failed to contact pgBackRest TLS server: no promary endpoint")
+            return False
```

Comment:
[Sometimes](https://github.com/canonical/postgresql-operator/actions/runs/6157787902/job/16709350681) there's an unhandled exception due to subprocess None parameter.

Reply:
`s/promary/primary/`?